front/src/modules/timesStops/hooks/useOutputTableData.tsx

```diff
@@ -0,0 +1,102 @@
+
+import type {
+  PathPropertiesFormatted,
+  SimulationResponseSuccess,
+} from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import { formatSuggestedOperationalPoints } from 'modules/pathfinding/utils';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import { ISO8601Duration2sec, datetime2sec, secToHoursString } from 'utils/timeManipulation';
+
+import useComputedMargins from './useComputedMargins';
+import { findNextScheduledPoint } from '../utils';
+
+function useOutputTableData(
+  simulatedTrain: SimulationResponseSuccess,
+  pathProperties: PathPropertiesFormatted,
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  selectedTrainSchedule: TrainScheduleResult,
+  startTime: string,
+  pathLength?: number
+) {
+  const scheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
+  const suggestedOperationalPoints: SuggestedOP[] = useMemo(
+    () =>
+      pathLength
+        ? formatSuggestedOperationalPoints(
+            pathProperties.operationalPoints,
+            pathProperties.geometry,
+            pathLength
+          )
+        : [],
+    [pathProperties.operationalPoints, pathProperties.geometry]
+  );
+
+  const pathStepsWithOpPointIndices = selectedTrainSchedule.path.map((pathStep) => {
+    const correspondingOpPointIndex = suggestedOperationalPoints.findIndex(
+      (sugOpPoint) =>
+        'uic' in pathStep &&
+        sugOpPoint.uic === pathStep.uic &&
+        sugOpPoint.ch === pathStep.secondary_code
+    );
+    return {
+      ...pathStep,
+      correspondingOpPointIndex,
+    };
+  });
+  const trainSchedulePathByUic = keyBy(pathStepsWithOpPointIndices, (pathStep) => {
+    if ('uic' in pathStep) {
+      return `${pathStep.uic}-${pathStep.secondary_code}`;
+    }
+    if ('operational_point' in pathStep) {
+      return pathStep.operational_point;
+    }
+    return '';
+  });
+
+  return suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
+    const opPoint = operationalPoints[sugOpIndex];
+    const nextOpPoint = findNextScheduledPoint(
+      operationalPoints,
+      pathStepsWithOpPointIndices,
+      sugOpIndex
+    );
+    const pathKey = `${sugOpPoint?.uic}-${sugOpPoint?.ch}`;
+    const pathStep = trainSchedulePathByUic[pathKey];
+
+    if (pathKey in trainSchedulePathByUic) {
+      const pathId = 'id' in pathStep ? pathStep.id : '';
+      const schedule = scheduleByAt[pathId];
+      const arrival = schedule?.arrival ? ISO8601Duration2sec(schedule.arrival!) : null;
+      const stopFor = schedule?.stop_for ? ISO8601Duration2sec(schedule.stop_for!) : '';
+      const departure = arrival && stopFor ? secToHoursString(arrival + stopFor) : null;
+      const [theoreticalMargin, realMargin, diffMargins] = useComputedMargins(
```

Comment:
Use a desconsctruction instead of relying on the position in the order.
```suggestion
      const { theoreticalMargin, realMargin, diffMargins } = useComputedMargins(
```
You can also store the value directly in marginsData and use it like this:
```
return {
    ...,
    ...marginsData
}
```

Reply:
done.
I think it’s nice to have a visual of the entire object being returned.

Reply:
As you prefer

Reply:
i did the `...marginsData` and `...scheduleData` eventually